lighthouse-core/gather/gatherers/seo/crawlable-links.js

```diff
@@ -13,19 +13,30 @@ class CrawlableLinks extends Gatherer {
    * @param {LH.Gatherer.PassContext} passContext
    * @return {Promise<LH.Artifacts['CrawlableLinks']>}
    */
-  afterPass(passContext) {
+  async afterPass(passContext) {
     const expression = `(function() {
       ${pageFunctions.getElementsInDocumentString}; // define function on page
+      const resolveURLOrNull = url => {
+        try { return new URL(url, window.location.href).href; }
+        catch (_) { return null; }
+      };
+
       const selector = 'a[href]:not([rel~="nofollow"])';
       const elements = getElementsInDocument(selector);
```

Comment:
how about

```js
getElementsInDocument(selector).filter(e => __ElementMatches.call(e, 'svg a'))
```

Reply:
not 100% sure what you're requesting, like we separate them in two different arrays? or just that we use `.matches` to replace the `instanceof` check?

Reply:
hmmm that was the idea but the fact that resolving these baseVal is so annoying makes it kinda not worth it.


actually how about

```js
const elements = getElementsInDocument(selector).filter(e => __ElementMatches.call(e, 'svg a'))
return elements.map(node => ({
  href: new __URL(node.getAttribute('href'), window.location.href).toString(),  // but with try/catch
  text: // ...
        // oh but then links in SVG don't have innerText
```
okay yeah this is all terrible.

Reply:
yeah I agree it's all terrible, I started with `node.innerText || node.textContent` but then got worried about potential accidental fallbacks with script/style tag contents when it's empty and whatnot, wanted to keep it as narrow as possible.